File: Chocolate Distribution Problem[5-August-2023]/Solution.cpp
//  >> C++ CODE <<

class Solution{
    public:
    long long findMinDiff(vector<long long> a, long long n, long long m){
        
        sort(a.begin(),a.end());
        
        long long ans=INT_MAX;
        
        for(int i=0;i+m-1<n;i++)
        {
            ans=min(ans,a[i+m-1]-a[i]);
        }
    
            return ans;
    }   
};

File: Chocolate Distribution Problem[5-August-2023]/Solution.java
//  >> JAVA CODE <<

class Solution
{
    public long findMinDiff (ArrayList<Integer> a, int n, int m)
    {
        Collections.sort(a);
        
        int ans=Integer.MAX_VALUE;
        
        for(int i=0;i+m-1<n;i++)
        {
            ans=Math.min(ans,a.get(i+m-1)-a.get(i));
        }
        
        return ans;
        
    }
}